### ISSUE TITLE
The Corretto distribution fails to find existing versions when using a task cache 

### ISSUE THREAD

**Issue by jamestoyer** [actions/setup-java#474]
**Description:**
When using self-hosted runners on GHES with a tool cache, setup-java fails to find the cached version of the `corretto` distribution. This is due to Corretto using non-semver version numbers, e.g. for Java 17 the version is `17.0.6.10.1`

**Task version:**
3.3 and greater

**Platform:**
- [x] Ubuntu
- [x] macOS
- [x] Windows

**Runner type:**
- [ ] Hosted
- [x] Self-hosted

**Repro steps:**
To repro, build a task cache with a given version of Java, e.g.:
```yaml
      - name: Setup Java
        uses: actions/setup-java@v3
        with:
          distribution: 'corretto'
          java-version: '17'
```

Once the task cache has been build, run a job with the above step that uses the task cache. When you run the job setup-java will fail to find the cached version and and attempt to reach out to the internet. In our case this is blocked by a firewall rule. 
<img width="1303" alt="Screenshot 2023-03-27 at 14 11 27" src="https://user-images.githubusercontent.com/380725/227947805-4cb215eb-5572-40ab-bdf7-132191ded10d.png">

To show that this is indeed an issue with the directory names on disk, the following can be done to make setup-java find it. In our case the Corretto Java version is `17.0.6.10.1`:

```yaml
      - name: Fix Java Dir
      run: mv /opt/hostedtoolcache/Java_Corretto_jdk/17.0.6.10.1 /opt/hostedtoolcache/Java_Corretto_jdk/17.0.6

      - name: Setup Java
        uses: actions/setup-java@v3
        with:
          distribution: 'corretto'
          java-version: '17'
```
The results of running this are as follows
<img width="1303" alt="Screenshot 2023-03-27 at 14 17 50" src="https://user-images.githubusercontent.com/380725/227948632-da27635b-e21e-4f0b-8e0f-501ecd6a9ba8.png">

**Expected behavior:**
The setup-java action should correctly find cached versions of Corretto on disk

**Actual behavior:**
The setup-java action fails to find cached versions of Corretto on disk and instead attempts to download a version from the internet.

**Comment by IvanZosimov** [comment]
Hi, @jamestoyer  👋 ! Thanks for the issue, we will investigate it and get back to you with updates!

**Comment by IvanZosimov** [comment]
Hi, @jamestoyer 👋 This PR should resolve the issue, it's already been merge into the main. So you can try how it works using:
```yml
- uses: actions/setup-java@main
```
The `main` branch can be used only as a preview of the features that will be added to the next release as it may be not stable. As soon as the next release will be created, I'll ping you.

**Comment by emiksk** [comment]
@IvanZosimov 
When the bug fix will be released?
I'm having trouble with the bug and hope it will be resolved in v3.

Thank you for your works!

**Comment by IvanZosimov** [comment]
Hi, @emiksk 👋 Sorry for delays, it will be released very soon.

**Comment by IvanZosimov** [comment]
Hi, @emiksk and @jamestoyer 👋 The [new version](https://github.com/actions/setup-java/releases/tag/v3.12.0) of the setup-java is released. Please, check it out.

I'm going to close this issue, If you have any additions or questions feel free to ping us.
